src/Common.am

```diff
@@ -73,3 +73,7 @@ COMPAT_LIB = $(top_builddir)/compat/libcompatsquid.la $(LIBPROFILER)
 
 ## Some helpers are written in Perl and need the local shell defined properly
 subst_perlshell = sed -e 's,[@]PERL[@],$(PERL),g' <$(srcdir)/$@.pl.in >$@ || ($(RM) -f $@ ; exit 1)
+
+# EXTENSIONS=.hdrtest
+# .h.hdrtest:
+# 	$(SHELL) $(top_srcdir)/test-suite/testheaders.sh "$(CXXCOMPILE)" $^ || exit 1
```

Comment:
> After doing some more tests and understanding the (quite cryptic) documentation, my proposed approach won't work. autotools really really wants to run the tests; the custom recipes build them, but that's not enough, they need to leave an artifact to be run which is "the test"

Sorry to hear that. Does that "proposed approach" mean/imply this $(SHELL) line? 

If the answer is yes: I do not know enough about Automake to be certain, but, AFAICT, this line contradicts the hdrtest_LOG_COMPILER declaration elsewhere in this PR -- if we are migrating to a LOG_COMPILER-based [approach](https://www.gnu.org/software/automake/manual/html_node/Parallel-Test-Harness.html), then we should let Automake(-generated code) call testheaders.sh (i.e. our "log compiler").

Otherwise: FWIW, I cannot reconcile "to leave an artifact to be run as the test" problem above and [documentation](https://www.gnu.org/software/automake/manual/html_node/Parallel-Test-Harness.html) quoted below. In that documentation, `wrapper-script -d baz`, for example, is executed automatically and does not seem to produce a "test to run". It produces test artifacts. That `wrapper-script -d baz` is the test. In our case, I would expect our (adjusted) testheaders.sh to be that `wrapper-script`.

----

```makefile
TESTS = foo.pl bar.py baz
TEST_EXTENSIONS = .pl .py
PL_LOG_COMPILER = $(PERL)
AM_PL_LOG_FLAGS = -w
PY_LOG_COMPILER = $(PYTHON)
AM_PY_LOG_FLAGS = -v
LOG_COMPILER = ./wrapper-script
AM_LOG_FLAGS = -d
```

... will invoke `$(PERL) -w foo.pl`, `$(PYTHON) -v bar.py`, and `./wrapper-script -d baz` to produce foo.log, bar.log, and baz.log, respectively. The foo.trs, bar.trs and baz.trs files will be automatically produced as a side-effect.